src/libraries/Microsoft.PowerFx.Interpreter/EvalVisitor.cs

```diff
@@ -300,38 +391,61 @@ public override async ValueTask<FormulaValue> Visit(CallNode node, EvalVisitorCo
                     args[i] = await child.Accept(this, context.IncrementStackDepthCounter()).ConfigureAwait(false);
                 }
                 else
-                {
+                {
+                    // $$$ This is where Lambdas are created! They close over key values to invoke.
                     args[i] = new LambdaFormulaValue(node.IRContext, child, this, context);
                 }
             }
 
             var childContext = context.SymbolContext.WithScope(node.Scope);
 
             FormulaValue result;
+
+            // [1] ??? who uses this?
+            // Config.EnableRegExFunctions --> Config.AdditionalFunctions
             IReadOnlyDictionary<TexlFunction, IAsyncTexlFunction> extraFunctions = _services.GetService<IReadOnlyDictionary<TexlFunction, IAsyncTexlFunction>>();
 
             try
             {
-                if (func is IAsyncTexlFunction asyncFunc || extraFunctions?.TryGetValue(func, out asyncFunc) == true)
+                var invoker = GetInvoker(func);
+
+                // $$$ New standard invoke path. Make everything go through here. 
+                // Eventually collapse all cases to this. 
+                if (invoker != null)

```

Comment:
This is the key change. 
Trying to take this complicated dispatch switch-case  and reduce it down to a single case. 

This iteration succesfully removes:
- IAsyncTexlFunction2
- IAsyncConnectorTexlFunction
- CustomTexlFunction
- IAsyncTexlFunctionJoin

And has adapters for
- UserDefinedFunction
- AsyncFunctionPtr

And a clear path to remove remaining ones.